MovieApp/MovieWithSummaryCell.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  MovieCell.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 04.04.2023..
+//
+
+import UIKit
+import PureLayout
+
+class MovieWithSummaryCell: UICollectionViewCell {
+    private var movieNameLabel: UILabel!
+    private var movieImage: UIImageView!
+    private var summaryLabel: UILabel!
+    private var cellButton: UIButton!
+    private var movieId: Int!
+    private var router: AppRouter!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    private func commonInit() {
+        assignViews()
+        style()
+        layout()
+    }
+    
+    func setData(imageUrl: String, name: String, summary: String, movieId: Int, router: AppRouter) {
+        movieNameLabel.text = name
+        self.movieId = movieId
+        self.router = router
+        Task {
+            await loadImage(imageURL: imageUrl, imageView: movieImage)
+        }
+        summaryLabel.text = summary
+    }
+    
+    private func assignViews() {
+        cellButton = UIButton()
+        cellButton.addTarget(self, action: #selector(handleCellButton), for: .touchUpInside)
+        contentView.addSubview(cellButton)
```

Comment:
This is applicable for the `MovieBannerCell` as well

Regarding one of my last comments, this should have been accomplished with a different approach

However, regarding this specific view:
The button is acting as it is a container view (UIVIew), rather than being a button. This could have been achieved by adding a `gesture recognizer` on the cell itself with the `UITapGestureRecgognizer` (you would also need to set the `isUserInteractionsEnabled` to true)

Note: I encourage you to check the other approaches: [didSelectItem](https://developer.apple.com/documentation/uikit/uicollectionviewdelegate/1618032-collectionview), [delegate pattern](https://medium.com/@armanabkar/delegate-pattern-in-ios-a26f2b329097#:~:text=A%20design%20pattern%20that%20enables%20a%20class%20or%20structure%20to,an%20instance%20of%20another%20type&text=Delegation%20is%20a%20design%20pattern,an%20instance%20of%20another%20type.), or a closure block